src/Microsoft.AspNetCore.DataProtection.Extensions/DataProtectionProvider.cs

```diff
@@ -49,13 +53,14 @@ public static IDataProtectionProvider Create(DirectoryInfo keyDirectory)
             // build the service collection
             var serviceCollection = new ServiceCollection();
             var builder = serviceCollection.AddDataProtection();
-            builder.PersistKeysToFileSystem(keyDirectory);
 
-            if (setupAction != null)
+            if (keyDirectory != null)
```

Comment:
This works? And falls through to registry, or azure directory or whatever? I'm kinda shocked. 


However, I'd still have expected null or as keyDirectory would throw an exception. In this method you would expect the parameter. 

So, refactor the service collection setup et al after to a private method, then put the argumentnullexception back and have both methods call into the private one.